sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,731 @@
+    (ElemsPerAddr == 1 || ElemsPerAddr == 2 || ElemsPerAddr == 4)),
+    void>::type
+flat_store(T *p, simd<T, n*ElemsPerAddr> vals, simd<uint32_t, n> offsets,
+           simd<uint16_t, n> pred = 1);
+```
+##### Flat-address block load/store
+read or write a consecutive block of data for the memory location specified by
+**addr**.
+
+```cpp
+template <typename T, int n, CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+simd<T, n> flat_block_load(const T *const addr);
+
+template <typename T, int n, CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+void flat_block_store(T *p, simd<T, n> vals);
+```
+
+##### Flat-address load4/store4
+perform scattered read/write for the given starting pointer **p** and
+**offsets**. Up to 4 data elements may be accessed at each address depending on
+the enabled channel **Mask**.
+
+```cpp
+template <typename T, int n, ChannelMaskType Mask,
+          CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+    typename std::enable_if<(n == 16 || n == 32),
+                            simd<T, n * __NumChannels(Mask)>>::type
+    flat_load4(T *p, simd<uint32_t, n> offsets, simd<uint16_t, n> pred = 1);
+
+template <typename T, int n, ChannelMaskType Mask,
+          CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+typename std::enable_if<(n == 16 || n == 32), void>::type
+    flat_store4(T *p, simd<T, n * __NumChannels(Mask)> vals,
+            simd<uint32_t, n> offsets, simd<uint16_t, n> pred = 1);
+```
+
+##### Flat-address atomic inc/dec
+perform atomic memory access operation with zero source operand.
+
+```cpp
+template <CmAtomicOpType Op, typename T, int n,
+          CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+    typename std::enable_if<__check_atomic<Op, T, n, 0>(), simd<T, n>>::type
+    flat_atomic(T *p, simd<unsigned, n> offset, simd<ushort, n> pred);
+```
+
+- ```Flat-address atomic add/sub/min/max/etc.```: perform atomic memory
+access operation with one source operand.
+
+```cpp
+template <CmAtomicOpType Op, typename T, int n,
+          CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+    typename std::enable_if<__check_atomic<Op, T, n, 1>(), simd<T, n>>::type
+    flat_atomic(T *p, simd<unsigned, n> offset, simd<T, n> src0,
+                simd<ushort, n> pred);
+```
+
+- ```Flat-address atomic CMPXCHG```: perform atomic memory access operation
+with two source operands.
+
+```cpp
+
+template <CmAtomicOpType Op, typename T, int n,
+          CacheHint L1H = CacheHint::Default,
+          CacheHint L3H = CacheHint::Default>
+    typename std::enable_if<__check_atomic<Op, T, n, 2>(), simd<T, n>>::type
+    flat_atomic(T *p, simd<unsigned, n> offset, simd<T, n> src0,
+                simd<T, n> src1, simd<ushort, n> pred);
+```
+
+#### Accessor-based memory access
```

Comment:
Lots of formatting errors in this section again (e.g. T  element type).

Reply:
thanks. pptx copy-paste problem